cfme/tests/configure/test_proxy.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+
+from cfme.cloud.provider import CloudProvider
+from cfme.common.provider import DefaultEndpointForm
+from cfme.utils import conf
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.providers import get_mgmt
+from cfme.utils.ssh import SSHClient
+from cfme.utils.virtual_machines import deploy_template
+from cfme.utils.wait import wait_for
+
+pytestmark = [
+    pytest.mark.tier(3),
+    pytest.mark.provider([CloudProvider], scope='module')
+]
+
+
+@pytest.yield_fixture(scope="module")
+def proxy_machine():
+    """ Deploy vm for proxy test
+
+    This fixture uses for deploy vm on provider from yaml and then receive it's ip
+    After test run vm deletes from provider
+    """
+    depot_machine_name = "test_proxy_{}".format(fauxfactory.gen_alphanumeric())
+    data = conf.cfme_data.get("proxy_template")
+    proxy_provider_key = data["provider"]
+    proxy_template_name = data["template_name"]
+    proxy_port = data['port']
+    prov = get_mgmt(proxy_provider_key)
+    deploy_template(proxy_provider_key,
+                    depot_machine_name,
+                    template_name=proxy_template_name)
+    yield prov.get_ip_address(depot_machine_name), proxy_port
+    prov.delete_vm(depot_machine_name)
+
+
+def validate_provider(provider, valid_proxy=True):
+    provider.refresh_provider_relationships()
+    wait_for(lambda: provider.last_refresh_error() is None, fail_condition=not valid_proxy,
+             timeout=200, delay=20)
+    result = provider.last_refresh_error() is None if valid_proxy else provider.last_refresh_error()
+    assert result
+
+
+def validate_provider_credentials(provider):
+    # need to validate credentials after incorrect proxy settings to be able to refresh provider
+    view = navigate_to(provider, 'Edit')
+    endp_view = provider.create_view(DefaultEndpointForm)
+    endp_view.validate.click()
+    view.cancel.click()
+
+
+def validate_proxy_logs(proxy_ip, appliance_ip, valid_proxy=True):
+    proxy_ssh = SSHClient(hostname=proxy_ip, **conf.credentials['proxy_vm'])
+    proxy_ssh.run_command('echo "" > /var/log/squid/access.log')
+
+    def _is_ip_in_log():
+        log_file = proxy_ssh.run_command("cat /var/log/squid/access.log").output
+        return appliance_ip in log_file
+    # need to wait until requests will occur in access.log or check if its empty after some time
+    wait_for(func=_is_ip_in_log, fail_condition=not valid_proxy, num_sec=180, delay=10)
+    assert _is_ip_in_log() == valid_proxy
+
+
+@pytest.yield_fixture(scope="function")
```

Comment:
REQUIRED: not needed anymore - kinda deprecated and replaced by  just `pytest.fixture`